### ISSUE TITLE
Documentation: haxe-language-server doesn't work until the first onDidChangeConfiguration event

### ISSUE THREAD

**Issue by joshtynjala** [vshaxe/vshaxe#359]
This is a request to document in the readme that there's a requirement to send a didChangeConfiguration notification as part of initializing the Haxe language server.

Making the notification optional would also be fine.

I ran into this issue trying to get the Haxe language server working in Moonshine IDE. Everything seemed to work up through the initialized notification, and then the language server didn't accept any further requests, like hover, definition or completion.

Luckily, I'm the one who implemented the client side of the protocol in Moonshine, so I can ensure that this notification is always sent. In my initial tests, adding this notification after init doesn't seem to hurt any existing language servers that I've previously integrated that worked fine without it.

**Comment by Gama11** [comment]
I think it's actually part of the LSP spec that there's an initial `onDidChangeConfiguration`?

**Comment by joshtynjala** [comment]
@Gama11 I'm not seeing that mentioned in the spec...

**Comment by Gama11** [comment]
I might be wrong, but then we should actually change this as it would be a violation of the spec, not document it. I think other language clients actually struggled with this too.

**Comment by joshtynjala** [comment]
Yeah, I figured out that this notification was needed based on the comments in #171 and #328 related to Neovim and Sublime.

**Comment by jdonaldson** [comment]
I ran into this as well, it required a lot of debugging and code tracing to figure out what was going wrong.

**Comment by kLabz** [comment]
We might want to rework this workflow, apparently there's `workspace/configuration` which could lead to less issues.

Adding this for reference, will look into this later: https://github.com/Microsoft/language-server-protocol/issues/567
